tests/test_freetextsearch.py

```diff
@@ -313,12 +1619,71 @@ def testBestHitSearchFieldsPostTarget(self):
         self.assertEqual(first_result_ens['approved_symbol'], 'BRAF')
     
         
-        eff_data = json_response['data'][5];
-        self.assertEqual(eff_data['q'], 'eff');
-        self.assertEqual(eff_data['exact'], False);
+        eff_data = json_response['data'][5]
+        self.assertEqual(eff_data['q'], 'eff')
+        self.assertEqual(eff_data['exact'], False)
         self.assertEqual(len(eff_data['data']),1) #should only get name and id
         self.assertEqual(eff_data['data']['approved_symbol'],'UBE2D3')
-        
+
+    def testBestHitSearchExactMatchManyTargetPost(self):
+        targets = GENE_SYMBOL_LIST
+        response = self._make_request('/api/latest/private/besthitsearch',
+                                      data=json.dumps({
+                                          'q': targets,
+                                          'filter': 'target',
+                                          'no_cache': True}),
+                                      content_type='application/json',
+                                      method='POST',
+                                      token=self._AUTO_GET_TOKEN)
+
+        self.assertTrue(response.status_code == 200)
+        json_response = json.loads(response.data.decode('utf-8'))
+
+        self.assertEqual(len(json_response['data']), len(targets))
+
+        for result in json_response['data']:
+            self.assertIsNotNone(result)
+
+    def testBestHitSearchExactMatchMart410TargetPost(self):
+        targets = TEST_MART_410
+        response = self._make_request('/api/latest/private/besthitsearch',
+                                      data=json.dumps({
+                                          'q': targets,
+                                          'filter': 'target',
+                                          'no_cache': True,
+                                      },),
+                                      content_type='application/json',
+                                      method='POST',
+                                      token=self._AUTO_GET_TOKEN)
+
+        self.assertTrue(response.status_code == 200)
+        json_response = json.loads(response.data.decode('utf-8'))
+
+        self.assertEqual(len(json_response['data']), len(targets))
+
+        for result in json_response['data']:
+            self.assertIsNotNone(result)
+
```

Comment:
I would love to test this new version you have made with webapp. Sorry, did not get a chance to do it today. If it works fine - may be we can also increase that number i had for multi search size from 200